test/integration/depth_camera.cc

```diff
@@ -458,6 +463,229 @@ void DepthCameraTest::DepthCameraBoxes(
   ignition::rendering::unloadEngine(engine->Name());
 }
 
+
+void DepthCameraTest::DepthCameraParticles(
+    const std::string &_renderEngine)
+{
+  int imgWidth_ = 256;
+  int imgHeight_ = 256;
+  double aspectRatio_ = imgWidth_/imgHeight_;
+
+  // box should fill camera view
+  // we will add particle emitter in between box and depth camera later
+  ignition::math::Vector3d boxSize(1.0, 10.0, 10.0);
+  ignition::math::Vector3d boxPosition(1.8, 0.0, 0.0);
+
+  // particle emitter is only supported in ogre2
+  if (_renderEngine.compare("ogre2") != 0)
+  {
+    igndbg << "Engine '" << _renderEngine
+              << "' doesn't support depth cameras" << std::endl;
+    return;
+  }
+
+  // Setup ign-rendering with an empty scene
+  auto *engine = ignition::rendering::engine(_renderEngine);
+  if (!engine)
+  {
+    igndbg << "Engine '" << _renderEngine
+              << "' is not supported" << std::endl;
+    return;
+  }
+
+  ignition::rendering::ScenePtr scene = engine->CreateScene("scene");
+
+  // red background
+  scene->SetBackgroundColor(1.0, 0.0, 0.0);
+
+  // Create an scene with a box in it
+  scene->SetAmbientLight(1.0, 1.0, 1.0);
+  ignition::rendering::VisualPtr root = scene->RootVisual();
+
+  // create blue material
+  ignition::rendering::MaterialPtr blue = scene->CreateMaterial();
+  blue->SetAmbient(0.0, 0.0, 1.0);
+  blue->SetDiffuse(0.0, 0.0, 1.0);
+  blue->SetSpecular(0.0, 0.0, 1.0);
+
+  // create box visual
+  ignition::rendering::VisualPtr box = scene->CreateVisual();
+  box->AddGeometry(scene->CreateBox());
+  box->SetOrigin(0.0, 0.0, 0.0);
+  box->SetLocalPosition(boxPosition);
+  box->SetLocalRotation(0, 0, 0);
+  box->SetLocalScale(boxSize);
+  box->SetMaterial(blue);
+  root->AddChild(box);
+  {
+    double farDist = 10.0;
+    double nearDist = 0.01;
+    double hfov_ = 1.05;
+    // Create depth camera
+    auto depthCamera = scene->CreateDepthCamera("DepthCamera");
+    ASSERT_NE(depthCamera, nullptr);
+
+    ignition::math::Pose3d testPose(ignition::math::Vector3d(0, 0, 0),
+        ignition::math::Quaterniond::Identity);
+    depthCamera->SetLocalPose(testPose);
+
+    // Configure depth camera
+    depthCamera->SetImageWidth(imgWidth_);
+    EXPECT_EQ(depthCamera->ImageWidth(),
+      static_cast<unsigned int>(imgWidth_));
+    depthCamera->SetImageHeight(imgHeight_);
+    EXPECT_EQ(depthCamera->ImageHeight(),
+      static_cast<unsigned int>(imgHeight_));
+    depthCamera->SetFarClipPlane(farDist);
+    EXPECT_NEAR(depthCamera->FarClipPlane(), farDist, DOUBLE_TOL);
+    depthCamera->SetNearClipPlane(nearDist);
+    EXPECT_NEAR(depthCamera->NearClipPlane(), nearDist, DOUBLE_TOL);
+    depthCamera->SetAspectRatio(aspectRatio_);
+    EXPECT_NEAR(depthCamera->AspectRatio(), aspectRatio_, DOUBLE_TOL);
+    depthCamera->SetHFOV(hfov_);
+    EXPECT_NEAR(depthCamera->HFOV().Radian(), hfov_, DOUBLE_TOL);
```

Comment:
Can we use `EXPECT_DOUBLE_EQ` instead of `EXPECT_NEAR`?

Reply:
963bc43